client/ayon_deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -517,107 +521,29 @@ def inject_ayon_environment(deadlinePlugin):
-            export_url
-        ]
-
-        # staging requires passing argument
-        # TODO could be removed when PR in ayon-core starts to fill
-        #  'AYON_USE_STAGING' (https://github.com/ynput/ayon-core/pull/1130)
-        #  - add requirement for "core>=1.1.1" to 'package.py' when removed
-        settings_variant = job.GetJobEnvironmentKeyValue(
-            "AYON_DEFAULT_SETTINGS_VARIANT"
-        )
-        if settings_variant == "staging":
-            args.append("--use-staging")
-
-        # Backwards compatibility for older versions
-        legacy_args = [
-            "--headless",
-            "extractenvironments",
-            export_url
-        ]
-
-        for key, value in add_kwargs.items():
-            args.extend(["--{}".format(key), value])
-            # Legacy arguments expect '--asset' instead of '--folder'
-            if key == "folder":
-                key = "asset"
-            legacy_args.extend(["--{}".format(key), value])
-
-        environment = {
-            "AYON_SERVER_URL": ayon_server_url,
-            "AYON_API_KEY": ayon_api_key,
-            "AYON_BUNDLE_NAME": ayon_bundle_name,
-        }
-
-        for key in ("AYON_USE_STAGING", "AYON_IN_TESTS"):
-            value = job.GetJobEnvironmentKeyValue(key)
-            if value:
-                environment[key] = value
-
-        for env, val in environment.items():
-            # Add the env var for the Render Plugin that is about to render
-            deadlinePlugin.SetEnvironmentVariable(env, val)
-            # Add the env var for current calls to `DeadlinePlugin.RunProcess`
-            deadlinePlugin.SetProcessEnvironmentVariable(env, val)
-
-        args_str = subprocess.list2cmdline(args)
-        print(">>> Executing: {} {}".format(exe, args_str))
-        process_exitcode = deadlinePlugin.RunProcess(
-            exe, args_str, os.path.dirname(exe), -1
-        )
-
-        if process_exitcode != 0:
-            print(
-                "Failed to run AYON process to extract environments. Trying"
-                " to use legacy arguments."
-            )
-            legacy_args_str = subprocess.list2cmdline(legacy_args)
-            process_exitcode = deadlinePlugin.RunProcess(
-                exe, legacy_args_str, os.path.dirname(exe), -1
-            )
-            if process_exitcode != 0:
-                raise RuntimeError(
-                    "Failed to run AYON process to extract environments."
+        output_dir = _get_output_dir(job)
+
+        worker_platform = platform.system().lower()
+        environment_file_name = f"extractenvironments_{worker_platform}.txt"
+        export_url = os.path.join(output_dir, environment_file_name)
+        _wait_for_in_progress(export_url)
+
+        if not os.path.exists(export_url):
+            print(f"'{export_url}' doesn't exist yet, extracting...")
+            temp_export_url = f"{export_url}.tmp"
+            with open(temp_export_url, "w"):
```

Comment:
Shouldn't this remove the temp file when it crashes?
```suggestion
            with open(temp_export_url, "w"):
                pass

            try:
                extractenvironments(
                    ayon_server_url,
                    ayon_api_key,
                    ayon_bundle_name,
                    deadlinePlugin,
                    exe,
                    temp_export_url,
                    job
                )
            except Exception:
                os.remove(temp_export_url)
            
            if (
                os.path.exists(temp_export_url)
                and not os.path.exists(export_url)
            ):
                print(f"Creating env var file {export_url}")
                shutil.move(temp_export_url, export_url)
```